### DISCUSSION TITLE
SQL Script: invalid character

### DISCUSSION THREAD

**Discussion by bikeridercz** [dbeaver/dbeaver#21985] [Q&A] [answered]
### Description

Dear friends, I decided to log this issue because it is annoying and happens quite often ;-) Probably it is caused by autocomplete feature that inserts invalid character between asterisk (*) and FROM keyword. When you look at a picture ... there seems to be a SPACE character in between, but it is NOT - there should be something else because execution of a command results with SQL Error. It is quite easy to fix it, you have to delete this character and make a real space instead, that's it. See the picture please & try to fix it ;-)

<img width="448" alt="image" src="https://github.com/dbeaver/dbeaver/assets/5019226/82efaca1-a80e-496a-866f-f626ae34ae0e">
 

### DBeaver Version

Community 23.1.3, july 24, 2023

### Operating System

macos 13.5.2

### Database and driver

Problem repeats on both - Oracle and MySQL driver.

### Steps to reproduce

see the picture

**Comment by uslss** [comment]
@bikeridercz , unfortunately, cannot reproduce the issue. It will be very helpful if you provide video with the issue.
Also, maybe it will be more clear, if you'll enable Show whitespace characters in Preferences
![image](https://github.com/dbeaver/dbeaver/assets/31996417/7205c6cb-d7e9-49a1-9064-05d9389201d1)

**Comment by E1izabeth** [comment]
It's been a while since no update here. If the issue is still actual please let me know and provide additional information.

**Comment by bikeridercz** [comment]
Sorry for delay, I cannot reproduce it, it seems to be occasional and it is still there (Version 23.2.2.202310081633) ... As I said before, it seems that autocomplete inserts something in between "select *" and "from", see the picture

<img width="783" alt="image" src="https://github.com/dbeaver/dbeaver/assets/5019226/1f4e8001-1921-4b6c-b6b7-a477f0d8e00d">

**Comment by E1izabeth** [comment]
@bikeridercz could you send this script as file? If it contains sensitive information you can made a duplicate of the file removing the other queries except the interested one.

**Comment by bikeridercz** [comment]
Sorry for delay but it takes some time the bug to appear, but here is a screenshot and sql script in attachment ... hope it helps ...

<img width="757" alt="image" src="https://github.com/dbeaver/dbeaver/assets/5019226/6727e21f-7cf2-40de-83b1-501909916eee">

BTW I was forced to rename .sql -> .sql.txt to upload it here, strange ;-)
[Script-10.sql.txt](https://github.com/dbeaver/dbeaver/files/13249908/Script-10.sql.txt)

**Comment by E1izabeth** [comment]
Thank you!

**Comment by E1izabeth** [comment]
Here is the reason for the issue 
![image](https://github.com/dbeaver/dbeaver/assets/28875055/f674ebbe-ecc0-4985-97df-2d0ec9cef37b)

You could try replacing this character by a whitespace, as you mentioned.

**Comment by E1izabeth** [comment]
However, I'm not sure how it could happen during autocompletion.

**Comment by E1izabeth** [comment]
@bikeridercz 
It looks like you accidentally pressed `Option+Space` instead of `Control+Space`. `Option+Space` inserts nsbp which is not supported by databases.

**Comment by bikeridercz** [comment]
Hi
- i do not need to replace it, it is easy to press backspace to delete wrong character and then make a space. after that the command can be executed successfuly
- i never press option+space or control+space, the wrong character is inserted automaticaly by dbeaver not by me

**Comment by E1izabeth** [comment]
Could you explain then how do you call autocompletion? We don't understand how it could be possible and how to reproduce it without pressing `Option+Space`

**Comment by bikeridercz** [comment]
OK, sorry then I do not want to say that the cause of the bug is "autocomplete", because I never use it, I never use option+space, I do not event know this shortcut.
Instead I'm saying that there should be some logic that during you typing makes a keywords uppercase (select -> SELECT) etc so somewhere over there is maybe the root cause ... I'll try to make some video, but this bug do not appear all the time, this is maybe the reason why you can't reproduce it on your side

**Comment by E1izabeth** [comment]
Maybe you don't use the shortcut, but do you choose the table name or column name from the proposed ones when typing? It's autocompletion too.

I hope you could notice some pattern of when it appears. Maybe it's reproducible only on some concrete database, or schema, or table.

We will be happy to see the video if you manage to record it.

Thanks

**Comment by bikeridercz** [comment]
here is a video, on second 12 bug is reproduced, on second 26 I press a backspace to remove wrong character and executed command once again with no problem.

https://github.com/dbeaver/dbeaver/assets/5019226/bcae3124-d16f-44c6-9f3b-e7871e7c49ac

**Comment by bikeridercz** [comment]
Oracle DB.

**Comment by E1izabeth** [comment]
Thank you!
We'll deep into it.

**Comment by E1izabeth** [comment]
Similar issue #14582

**Comment by LonwoLonwo** [comment]
1. Could you please try to disable auto-completion and try again?

![2023-11-16 16_19_39-Transparent Window](https://github.com/dbeaver/dbeaver/assets/45152336/93c07568-63ad-4feb-8938-2932e4f36bcf)

2. How did you type the symbol "*"? Which tabs did you use? Can be related to [this problem](https://superuser.com/questions/416022/hidden-characters-inserted-after-pipe-followed-by-a-space)

**Comment by bikeridercz** [comment]
ad 1) OK, I will try it
ad 2) on mac keyboard I have to press Option+8

**Comment by bikeridercz** [comment]
> Similar issue #14582

no, there is no similarity, on mentioned issue the SQL command is copied from outside of DBeaver, so the wrong non visible character comes probably from outside. I my case, I'm typing the sql command by hand directly in DBeaver so that the wrong non visible character is inserted into sql text by DBeaver logic, somehow ...

**Comment by E1izabeth** [comment, marked as answer]
If you type `*` with `Option+8`, then could you pay attention to the way you type `Space` after that? Is it possible that you type so fast that your `Space` become combined with `Option`? The character you have can be typed exactly by this keys combination (`Option+Space`).
Here is a way how to disable this keys combination https://superuser.com/questions/78245/how-to-disable-the-option-space-key-combination-for-non-breaking-spaces Could you try this solution?

**Comment by bikeridercz** [comment]
ah :-( yes you are right, this is the reason. i have to apologise for the time you spent with this issue. sorry, thanks for help.

**Comment by E1izabeth** [comment]
Thank you for checking! I'm closing this issue then.
